Analytics/Internal/SEGUtils.m

```diff
@@ -100,3 +112,163 @@ BOOL isUnitTesting()
     });
     return _isUnitTesting;
 }
+
+NSString *deviceTokenToString(NSData *deviceToken)
+{
+    if (!deviceToken) return nil;
+    
+    const unsigned char *buffer = (const unsigned char *)[deviceToken bytes];
+    if (!buffer) {
+        return nil;
+    }
+    NSMutableString *token = [NSMutableString stringWithCapacity:(deviceToken.length * 2)];
+    for (NSUInteger i = 0; i < deviceToken.length; i++) {
+        [token appendString:[NSString stringWithFormat:@"%02lx", (unsigned long)buffer[i]]];
+    }
+    return token;
+}
+
+NSString *getDeviceModel()
+{
+    size_t size;
+    sysctlbyname("hw.machine", NULL, &size, NULL, 0);
+    char result[size];
+    sysctlbyname("hw.machine", result, &size, NULL, 0);
+    NSString *results = [NSString stringWithCString:result encoding:NSUTF8StringEncoding];
+    return results;
+}
+
+BOOL getAdTrackingEnabled(SEGAnalyticsConfiguration *configuration)
+{
+    BOOL result = NO;
+    if ((configuration.adSupportBlock != nil) && (configuration.enableAdvertisingTracking)) {
+        result = YES;
+    }
+    return result;
+}
+
+NSDictionary *getStaticContext(SEGAnalyticsConfiguration *configuration, NSString *deviceToken)
+{
+    NSMutableDictionary *dict = [[NSMutableDictionary alloc] init];
+
+    dict[@"library"] = @{
+        @"name" : @"analytics-ios",
+        @"version" : [SEGAnalytics version]
+    };
+
+    NSMutableDictionary *infoDictionary = [[[NSBundle mainBundle] infoDictionary] mutableCopy];
+    [infoDictionary addEntriesFromDictionary:[[NSBundle mainBundle] localizedInfoDictionary]];
+    if (infoDictionary.count) {
+        dict[@"app"] = @{
+            @"name" : infoDictionary[@"CFBundleDisplayName"] ?: @"",
+            @"version" : infoDictionary[@"CFBundleShortVersionString"] ?: @"",
+            @"build" : infoDictionary[@"CFBundleVersion"] ?: @"",
+            @"namespace" : [[NSBundle mainBundle] bundleIdentifier] ?: @"",
+        };
+    }
+
+    UIDevice *device = [UIDevice currentDevice];
+
+    dict[@"device"] = ({
+        NSMutableDictionary *dict = [[NSMutableDictionary alloc] init];
+        dict[@"manufacturer"] = @"Apple";
+        dict[@"type"] = @"ios";
+        dict[@"model"] = getDeviceModel();
+        dict[@"id"] = [[device identifierForVendor] UUIDString];
+        dict[@"name"] = [device model];
+        if (getAdTrackingEnabled(configuration)) {
+            NSString *idfa = configuration.adSupportBlock();
+            // This isn't ideal.  We're doing this because we can't actually check if IDFA is enabled on
+            // the customer device.  Apple docs and tests show that if it is disabled, one gets back all 0's.
+            BOOL adTrackingEnabled = (![idfa isEqualToString:@"00000000-0000-0000-0000-000000000000"]);
```

Comment:
Should we plan for a different format than one with the dashes? Or just react if this ever breaks for some devices / iOS versions?

Reply:
I'm working off the docs so I think it's ok.  I don't like it all that much, but it makes a good deal of lift on customers if we require them to handle both.